.travis.yml

```diff
@@ -1,41 +1,160 @@
-os:
-  - linux
-  - osx
+#  2020-05-26: NOTE: Enabling experimental Travis feature of YAML check inside
+# Look into: Build #NUM -> Job #NUM.N -> View config -> Build config validation
+version: ~> 1.0
+
+# NOTE: Please, be aware that Travis YAML & docs & API are hard to make work properly. Travis configuration requires a lot of retries and some compromises. Tere are many ways that may look like that can be done in that way, but it would not work most of the time, or not work the way that you expect, need it. Travis config works only the certain particular ways. Some things look possible - but they are impossible in Travis. Current configuration is "the best way possible" that was found in ~100-150-200 retries, depending on what concider a retry.
+
+# NOTE: Let the official Travis YAML checker help you: https://config.travis-ci.com/explore
 
 language: nix
 
-sudo: required
+sudo: required    #  2020-05-26: NOTE: Despite deprecated, but siletly still RAM 4GB -> 7.5GB
 
 git:
+  quiet: true # NOTE: Do not log
   depth: 4    # NOTE: "The use of clone depth: 1 often results in a git error
               # when a new commit has been pushed to a branch before the CI
               # platform started cloning the intended commit."
 
+# NOTE: {os} x {jobs} - {exclude} = {build matrix}
+os:
+  - linux
+  - osx
+
+#  2020-05-26: NOTE: Currently newest Travis dist Ubuntu 18.04 bionic
+dist: bionic
+
+# #  2020-05-26: NOTE: Currently newest macOS image
+# osx_image: xcode11.4  # NOTE: Official Nix installer fails spectacularly on it.
+
 env:
   global:
-    - ALL_TESTS=yes
+    # NOTE: This is secure CACHIX_SIGNING_KEY=val
     - secure: "dm6I+M4+V+C7QMTpcSADdKPE633SvmToXZrTbZ7miNDGmMN+/SfHeN2ybi1+PW6oViMlbPN/7J/aEfiGjSJI8vLk72Y4uCWGmpSb8TXZLu6+whnxtZzzW8+z4tsM4048QJg7CF3N/25U8thRFgs3DqUub1Sf3nG9LrNWdz6ZcDQ="
+    # NOTE: Nix by default uses nixpkgs-unstable channel
+    # NOTE: Setup for Nixpkgs revision
+    # `rev` vals in order of freshness -> cache & stability (branch causes differ):
+    # { master
+    # , commitHash
+    # , haskell-updates
+    # , nixpkgs-unstable
+    # , nixos-unstable
+    # , nixos-20.03
+    # }
+    # - rev=nixos-unstable
+    # # NOTE: Switching into Nixpkgs revision
+    # - NIX_PATH="nixpkgs=https://github.com/nixos/nixpkgs/archive/$rev.tar.gz"
+    # NOTE: Project/binary name
+    - name=hnix
+    # NOTE: Used in the `generateOptparseApplicativeCompletions = true`
```

Comment:
Literally used in the case (literally inside the case) when `generateOptparseApplicativeCompletions = true`.

Since that variable used literally and it composed into the result - it is not used _when_, it is used _in the_ true case.

Reply:
AKA _in the_ is reference to look in the code in the code of the generateOptparseApplicativeCompletions = true case.

Reply:
```suggestion
    # NOTE: Used in the `generateOptparseApplicativeCompletions = true` case.
```